output/transform/postgres_statements.go

```diff
@@ -67,11 +67,27 @@ func transformQueryStatistic(stats state.DiffedPostgresStatementStats, idx int32
 	}
 }
 
-func transformPostgresStatements(s snapshot.FullSnapshot, newState state.PersistedState, diffState state.DiffState, transientState state.TransientState, roleOidToIdx OidToIdx, databaseOidToIdx OidToIdx) snapshot.FullSnapshot {
+type postgresStatementKey struct {
+	databaseOid state.Oid
+	userOid     state.Oid
+	queryID     int64
+}
+type StatementKeyToIdx map[postgresStatementKey]int32
+
+func transformPostgresStatements(s snapshot.FullSnapshot, newState state.PersistedState, diffState state.DiffState, transientState state.TransientState, roleOidToIdx OidToIdx, databaseOidToIdx OidToIdx) (snapshot.FullSnapshot, StatementKeyToIdx) {
 	// Statement stats from this snapshot
 	groupedStatements := groupStatements(transientState.Statements, diffState.StatementStats)
+	statementKeyToIdx := make(StatementKeyToIdx)
 	for key, value := range groupedStatements {
 		idx := upsertQueryReferenceAndInformation(&s, transientState.StatementTexts, roleOidToIdx, databaseOidToIdx, key, value)
+		for _, queryId := range value.queryIDs {
+			sKey := postgresStatementKey{
+				databaseOid: key.databaseOid,
+				userOid:     key.userOid,
+				queryID:     queryId,
```

Comment:
As discussed with Keiko in our 1:1, this currently does not keep the top-level identifier here - this is intentional because we currently do not send top_level to the server side. When we do so in the future we should expand this logic here as well to include it.